activity/aws-stepfunctions-activity.json

```diff
@@ -64,7 +64,8 @@
     },
     "read": {
       "permissions": [
-        "states:DescribeActivity"
+        "states:DescribeActivity",
+        "states:ListTagsForResource"
```

Comment:
will customers who don't have this permission today start to see failures if the role they use does not have the permission?

I understand this is about compliance, but asking whether it's breaking in any case

Reply:
According to [Developing resource types documentation](https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-develop.html):
```
If your resource type calls AWS APIs in any of its handlers, you must create an IAM execution role that includes the necessary permissions to call those AWS APIs, and provision that execution role in your account. CloudFormation then assumes that execution role to provide your resource type with the appropriate credentials.

When you call generate, the CloudFormation CLI automatically generates an execution role template, resource-role.yaml, as part of generating the code files for the resource type package. This template is based on the permissions specified for each handler in the handlers section of the Resource type schema. When you use submit to register the resource type, the CloudFormation CLI attempts to create or update an execution role based on the template, and then passes this execution role to CloudFormation as part of the registration.
```

Reply:
The execution role should update automatically since we updated the permissions for the Read handler in the resource schema.

Reply:
> you must create an IAM execution role that includes the necessary permissions to call those AWS APIs

I totally understand that, but as we are changing what "necessary permissions" entails here, something that used to work might not for a customer that is not leveraging the CloudFormation CLI. Confirming that this is a breakage we are consciously making.

Reply:
Yes, confirming that we were not returning the properties that were expected based on the resource handler contract, and that returning those expected properties requires additional permissions which may require customer roles to be updated.

Reply:
Update - The implementation has been refactored to be backwards compatible. If ListTagsForResource permissions are not provided, the request will not fail, but tags will not be included in the returned model.